src/main/cpp/_nix_based/jssc.cpp

```diff
@@ -527,27 +535,70 @@ JNIEXPORT jboolean JNICALL Java_jssc_SerialNativeInterface_writeBytes
 /*
  * Reading data from the port
  *
- * Rewrited in 2.5.0 (using select() function for correct block reading in MacOS X)
+ * Rewritten to use poll() instead of select() to handle fd>=1024
  */
 JNIEXPORT jbyteArray JNICALL Java_jssc_SerialNativeInterface_readBytes
-  (JNIEnv *env, jobject object, jlong portHandle, jint byteCount){
-    fd_set read_fd_set;
+  (JNIEnv *env, jobject thisObj, jlong portHandle, jint byteCount){
+
+    // Error-Handling:
+    // I have no idea why we just ignored any kind of errors (eg ignoring return
+    // values of 'select' and 'read') in the old code. I also could not find any
+    // comments which would explain why. I liked to communicate those by raising
+    // java exceptions. But this would not be backward compatible. So I restrict
+    // myself to just log the irregular cases for now. If I get some time, I might
+    // open another PR to add some error-handling. So we could discuss that topic
+    // there.
```

Comment:
You can comment on your own PR lines to explain things like these (or open as new issues).  The code should only include the appropriate `TODO`'s.